zerver/views/saved_snippets.py

```diff
@@ -45,6 +48,32 @@ def create_saved_snippet(
     return json_success(request, data={"saved_snippet_id": saved_snippet.id})
 
 
+@typed_endpoint
+def edit_saved_snippet(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    saved_snippet_id: PathOnly[int],
+    title: Annotated[
+        str | None,
+        StringConstraints(
+            min_length=1, max_length=SavedSnippet.MAX_TITLE_LENGTH, strip_whitespace=True
+        ),
+    ] = None,
+    content: Annotated[
+        str | None,
+        StringConstraints(
+            min_length=1, max_length=settings.MAX_MESSAGE_LENGTH, strip_whitespace=True
+        ),
+    ] = None,
+) -> HttpResponse:
+    if title is None and content is None:
+        raise JsonableError(_("No new data is supplied"))
```

Comment:
Do we have an existing pattern for "no changes detected"? I feel like it's not worth an extra string for this, but couldn't find an example in 20 seconds of looking.

It might be best to just delete this check and have it be a noop 200 if there's no changes; I think some other endpoints might do that.

Can be fixed in a follow-up if needed.